### ISSUE TITLE
Concatted strings not accepted in template argument of annotation

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#110]
```
@Component(template: 'a' + 'b'...)
...
```

and

```
@Component(template: 'a'
'b', ...)
...
```

both generate the error "A string value expected"

**Comment by MichaelRFairhurst** [comment]
This also does not work in selectors as threw us off on #86

**Comment by parren-google** [comment]
Also does not work with r'' r''.

**Comment by MichaelRFairhurst** [comment]
Just realize this is actually a good deal trickier to solve than my initial impression. We have to track offsets through this correctly.

Its simple enough in the mentioned examples:

```
  template: 'valid template stuff' + ' {{invalid}} template stuff',
```

we can hack this by setting offset to the contentOffset of 'valid template stuff`, and then as we calculate the constant value of the expression we can pad spaces to account for the gap between `stuff'` and `' {{invalid}}` so that the mustache is highlighted correctly. 

However constant expressions in general here offer a lot more problems. It may involve constant variables...

```
  const String myConstVar = '{{invalid}}';
  ...
    template: myConstVar + ' more stuff ' + myConstVar,
```

they can appear multiple times, have lower offsets than template text that's earlier in the static string, and they can even appear in different files!

We _can_ support this by having an offset map that's built alongside calculating the static value. Although that won't support separate files...that requires hacking into the work manager, I believe, because now file A can depend on file B but the errors in file B depend on file A in this case creating a circular dependency (which is resolvable, but nasty).

Even handling the simplest cases of padding line breaks and such is not trivial.

**Comment by matanlurey** [comment]
Do we even support con-catted strings for the compiler?

**Comment by MichaelRFairhurst** [comment]
Yes it does work, since it can be resolved into a string value. I think
angular then reports errors by printing the value and highlighting the
offsets. Problem is we can't print the highlighted value.

**Comment by MichaelRFairhurst** [comment]
Merged #164
